extensions/rvadapter/src/main/java/com/google/android/agera/rvadapter/RepositoryAdapter.java

```diff
@@ -76,6 +77,8 @@ public static Builder repositoryAdapter() {
     @NonNull
     final List<RepositoryPresenter<Object>> presenters = new ArrayList<>();
     @NonNull
+    final List<Boolean> isDynamicPresenter = new ArrayList<>();
```

Comment:
Instead of this array/list, let's upgrade RepositoryPresenter with a private (package-local) method:

```
  long getOffsetItemId(@NonNull final T data, final int index, final int offset) {
    final long rawItemId = getItemId(data, index);
    if (rawItemId == RecyclerView.NO_ID) { // NOTE THIS SPECIAL TREATMENT
      return rawItemId;
    }
    return rawItemId + offset;
  }
```

Then override this in your new class:

```
  private static final class LayoutRepositoryPresenter extends RepositoryPresenter<Object> {
    private final LayoutPresenter presenter;
    private final int sequenceNumber;  // NEW FIELD

    LayoutRepositoryPresenter(@NonNull final LayoutPresenter presenter, final int sequenceNumber) {
      this.presenter = presenter;
      this.sequenceNumber = sequenceNumber;
    }

    ...

    @Override
    long getOffsetItemId(@NonNull final Object data, final int index, final int offset) {
      return sequenceNumber;
    }
  }
```

And use it directly in the outer getItemId():
```
return presenters[resolvedRepositoryIndex].getOffsetItemId(
    data[resolvedRepositoryIndex], resolvedItemIndex,
    layoutPresenterCount);  // SPOT THE DIFFERENCE
```